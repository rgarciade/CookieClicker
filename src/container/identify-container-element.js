import {LitElement, html, css} from "lit";
//import {navigate} from "../components/nav-element.js";
//import {scores} from "/src/status/index.js";
import "../components/icon-element.js";
import "../components/button-element.js";
import {identifyStyle} from "./styles/identify-container-style.js";
import {commonStyle} from "../styles/common.js";
import {ScoresState} from "../status/index.js";

export class IdentifyContainerElement extends ScoresState(LitElement) {
    static get styles() {
        return [identifyStyle,commonStyle,
            css`
                input {
                    min-width: 400px;
                }
            `
        ];
    }
    constructor() {
        super();
        this.gamerName = '';
        this.haveError = false;

    }
    getScoresHtml(){
        const scoresData = this._scores_getScoresSorted();
        if(scoresData.length === 0) return html`<p>No scores yet</p>`;

        const htmlList = Object.keys(scoresData).slice(0, 4).sort(
            (a, b) => scoresData[b] - scoresData[a]
        ).map((order) => {
            const score = scoresData[order].value;
            const name = scoresData[order].playerName;
            return html`<li>${name}: ${score}</li>`
        });
        return html`<ol>${htmlList}</ol>`;
    }
    IsErrorName(){
       return  this.haveError ? html`<span class="error-span" >*Escribe un nombre correcto</span>` : ''
    }

    render() {
        return html`
            <div class="title" >
                <h2><icon-element icon="star" color="blue" size="60px"></icon-element>The Bests</h2>
            </div>
            
            <div class="container">
                <div class="bests" >
                    ${this.getScoresHtml()}
                </div>
                <input placeholder="Gamer Name" @input="${this._handleInput}"/>
                ${this.IsErrorName()}
                <button-element text="Join game" @click="${this._joinGame}">Play</button-element>
            </div>
        `;
    }

    _handleInput(event) {
        this.gamerName = event.target.value;
    }
    _joinGame(){
        if(!this.gamerName){

            const oldVal = this.haveError;
            this.haveError = true;
            this.requestUpdate('haveError',oldVal);
            return

        }
        this.dispatchEvent(new CustomEvent('router-navigate', {
            detail: '/game?name=' + encodeURIComponent(this.gamerName),
            bubbles: true,
            composed: true
        }));
    }
}


window.customElements.define('identify-container-element', IdentifyContainerElement);
