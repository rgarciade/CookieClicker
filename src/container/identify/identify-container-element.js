import { LitElement, html } from 'lit';
import '../../components/icon/icon-element.js';
import '../../components/button/button-element.js';
import { identifyStyle } from './identify-container-style.js';
import { commonStyle } from '../../styles/common.js';
import { ScoresState } from '../../status/index.js';

export class IdentifyContainerElement extends ScoresState(LitElement) {
	static get styles() {
		return [identifyStyle, commonStyle];
	}
	constructor() {
		super();
		this.gamerName = '';
		this.haveError = false;
	}
	getScoresHtml() {
		const scoresData = this._scores_getScoresSorted();
		if (scoresData.length === 0) return html`<p>No scores yet</p>`;

		const htmlList = Object.keys(scoresData)
			.slice(0, 4)
			.sort((a, b) => scoresData[b] - scoresData[a])
			.map((order) => {
				const score = scoresData[order].value;
				const name = scoresData[order].playerName;
				return html`<li>${name}: ${score}</li>`;
			});
		return html`<ol>
			${htmlList}
		</ol>`;
	}
	IsErrorName() {
		return this.haveError
			? html`<span class="error-span">*Escribe un nombre correcto</span>`
			: '';
	}

	render() {
		return html`
			<div class="title">
				<h2>
					<icon-element icon="star" class="icon-start"></icon-element>
					The Bests
				</h2>
			</div>

			<div class="container">
				<div class="bests">${this.getScoresHtml()}</div>
				${this.IsErrorName()}
				<input placeholder="Gamer Name" @input="${this._handleInput}" />
				<button-element
					class="play-button"
					text="Join game"
					@click="${this._joinGame}"
				>
					Play
				</button-element>
			</div>
		`;
	}

	_handleInput(event) {
		this.gamerName = event.target.value;
	}
	_joinGame() {
		if (!this.gamerName) {
			const oldVal = this.haveError;
			this.haveError = true;
			this.requestUpdate('haveError', oldVal);
			return;
		}
		// en game-container e usado otra forma de navegar, para probar varios conceptos
		this.dispatchEvent(
			new CustomEvent('router-navigate', {
				detail: '/game?name=' + encodeURIComponent(this.gamerName),
				bubbles: true,
				composed: true,
			})
		);
	}
}

window.customElements.define(
	'identify-container-element',
	IdentifyContainerElement
);
