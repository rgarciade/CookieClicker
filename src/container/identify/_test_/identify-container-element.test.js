import { html, fixture } from '@open-wc/testing';
import '../identify-container-element.js';
import { expect } from '@jest/globals';
import { cleanedLitElements } from '../../../../testFunctions/cleanedLit.js';
import { getAllElementsAndShadowRoots } from 'shadow-dom-testing-library';

describe('identify-container-element', () => {
	it('identify-container matches the snapshot', async () => {
		const el = await fixture(
			html`<identify-container-element></identify-container-element>`
		);
		const data = getAllElementsAndShadowRoots(el, '*');
		const dataCleaned = cleanedLitElements(data);
		expect(dataCleaned).toMatchSnapshot();
	});
});
