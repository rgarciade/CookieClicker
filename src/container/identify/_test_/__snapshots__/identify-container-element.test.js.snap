// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`identify-container-element identify-container matches the snapshot 1`] = `
[
  "[36m<identify-container-element />[39m",
  "ShadowRoot {}",
  "[36m<div[39m
  [33mclass[39m=[32m"title"[39m
[36m>[39m
  [0m
				[0m
  [36m<h2>[39m
    [0m
					[0m
    [36m<icon-element[39m
      [33mclass[39m=[32m"icon-start"[39m
      [33micon[39m=[32m"star"[39m
    [36m/>[39m
    [0m
					The Bests
				[0m
  [36m</h2>[39m
  [0m
			[0m
[36m</div>[39m",
  "[36m<h2>[39m
  [0m
					[0m
  [36m<icon-element[39m
    [33mclass[39m=[32m"icon-start"[39m
    [33micon[39m=[32m"star"[39m
  [36m/>[39m
  [0m
					The Bests
				[0m
[36m</h2>[39m",
  "ShadowRoot {}",
  "[36m<i[39m
  [33mclass[39m=[32m"material-icons"[39m
[36m>[39m
  [0mstar[0m
[36m</i>[39m",
  "[36m<style>[39m
  [0m
	:host {
		--mdc-icon-size: 24px;
	}
	.material-icons {
		font-family: 'Material Icons';
		font-weight: normal;
		font-style: normal;
		font-size: var(--icon-size, 120px); /* Preferred icon size */
		display: inline-block;
		line-height: 1;
		text-transform: none;
		letter-spacing: normal;
		word-wrap: normal;
		white-space: nowrap;
		direction: ltr;
		color: var(--icon-color, white);

		/* Support for all WebKit browsers. */
		-webkit-font-smoothing: antialiased;
		/* Support for Safari and Chrome. */
		text-rendering: optimizeLegibility;

		/* Support for Firefox. */
		-moz-osx-font-smoothing: grayscale;

		/* Support for IE. */
		font-feature-settings: 'liga';
	}
[0m
[36m</style>[39m",
  "[36m<div[39m
  [33mclass[39m=[32m"container"[39m
[36m>[39m
  [0m
				[0m
  [36m<div[39m
    [33mclass[39m=[32m"bests"[39m
  [36m>[39m
    [36m<p>[39m
      [0mNo scores yet[0m
    [36m</p>[39m
  [36m</div>[39m
  [0m
				[0m
  [0m
				[0m
  [36m<input[39m
    [33mplaceholder[39m=[32m"Gamer Name"[39m
  [36m/>[39m
  [0m
				[0m
  [36m<button-element[39m
    [33mclass[39m=[32m"play-button"[39m
    [33mtext[39m=[32m"Join game"[39m
  [36m>[39m
    [0m
					Play
				[0m
  [36m</button-element>[39m
  [0m
			[0m
[36m</div>[39m",
  "[36m<div[39m
  [33mclass[39m=[32m"bests"[39m
[36m>[39m
  [36m<p>[39m
    [0mNo scores yet[0m
  [36m</p>[39m
[36m</div>[39m",
  "[36m<p>[39m
  [0mNo scores yet[0m
[36m</p>[39m",
  "[36m<input[39m
  [33mplaceholder[39m=[32m"Gamer Name"[39m
[36m/>[39m",
  "ShadowRoot {}",
  "[36m<button[39m
  [33mclass[39m=[32m"primary long"[39m
[36m>[39m
  [0m
				[0m
  [36m<slot />[39m
  [0m
			[0m
[36m</button>[39m",
  "[36m<slot />[39m",
  "[36m<style>[39m
  [0m
	button {
		border-radius: 12px;
		background-color: var(--color-primario);
		border: none;
		color: white;
		padding: 15px 32px;
		text-align: center;
		text-decoration: none;
		display: inline-block;
		font-size: var(--button-font-size, 76px);
		margin: 4px 2px;
		cursor: pointer;
		width: var(--button-width, '2px');
		min-width: var(--button-width, '2px');
		height: var(--button-height, '2px');
	}
	button:disabled {
		background-color: grey;
	}
	.primary {
		background-color: var(--color-primario, blue);
	}
	.secondary {
		background-color: var(--color-secundario, red);
	}
	.long {
		width: 100%;
	}
	.short {
		width: 50%;
	}
[0m
[36m</style>[39m",
  "[36m<style>[39m
  [0m
	.container {
		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: center;
		height: 55vh;
		font-size: 56px;
		gap: 80px;
	}
	input {
		min-width: 7em;
	}
	.bests {
		min-height: 34vh;
	}
	.title {
		font-size: 110px;
		margin-bottom: 0px;
		padding-bottom: 0px;
		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: center;
	}
	.title h2 {
		display: flex;
		align-items: center;
		margin-bottom: 0px;
	}

	.error-span {
		font-size: 40px;
		color: red;
		font-weight: 600;
	}
	ol {
		font-size: 70px;
		line-height: 1.5;
	}
	.play-button {
		--button-width: 5.7em;
	}
	.icon-start {
		--icon-color: red;
	}
	@media only screen and (max-width: 1080px) {
		.title {
			font-size: 50px;
		}
		ol {
			font-size: 49px;
			padding-top: 1em;
		}
	}
    @media only screen and (max-width: 550px) {
        .title {
            font-size:30px;
            --icon-size: 50px;
        }
        ol {
            font-size: 30px;
            padding-top: 3em;
        }
        input {
            font-size: 26px !important;
            min-width: 11em;
        }
        .play-button {
            --button-width: 4.3em;
        }
    }
[0m
[36m</style>[39m",
  "[36m<style>[39m
  [0m
	input {
		font-size: 20px;
		padding: 10px;
		border-radius: 5px;
		width: 70px;
	}
	input {
		font-size: 56px;
		padding: var(--spacing, 20px);
		border-radius: 12px;
		border: 1px solid black;
	}
	h1 {
		font-size: 56px;
	}
	select {
		font-size: 36px;
	}
[0m
[36m</style>[39m",
]
`;
